### PR TITLE
Add additional openai request parameters to config and chat_completion

### PR DESCRIPTION
**Pull request by Enigma94**
### Background

In case of the need to fine-tune the model, not only with temperature but also other parameters such as top_p.

### Changes

Add configurability of the OpenAI chat completion parameters by implementing them into the configuration file.

### Test Plan

- Changes config parameters
- Added parameters using create_chat_completion

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — DISMISSED** [review]

File: .env.template [hunk 1]
```diff
@@ -2,6 +2,10 @@ PINECONE_API_KEY=your-pinecone-api-key
 PINECONE_ENV=your-pinecone-region
 OPENAI_API_KEY=your-openai-api-key
 TEMPERATURE=1
+TOP_P=1
```
Comment (richbeales) [inline .env.template]: Can you please prefix these new keys OPENAI_ ?

Comment (Enigma94) [reply .env.template]: Sure, i actually did it at first but remove it for some reason

Comment (Enigma94) [reply .env.template]: Please approve asap, as I dont want to keep solving merge conflicts..

**Review by nponeccop — APPROVED** [review]

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@Enigma94 There are conflicts now

**Pull request closed without merging**